packages/hardhat/contracts/Ethernauts.sol

```diff
@@ -162,36 +175,47 @@ contract Ethernauts is ERC721Enumerable, Ownable {
         _tokensGifted += 1;
     }
 
-    function setMintPrice(uint newMintPrice) external onlyOwner {
+    function setMintPrice(uint256 newMintPrice) external onlyOwner {
         mintPrice = newMintPrice;
+        emit MintPriceChanged(newMintPrice);
     }
 
-    function setEarlyMintPrice(uint newEarlyMintPrice) external onlyOwner {
+    function setEarlyMintPrice(uint256 newEarlyMintPrice) external onlyOwner {
         earlyMintPrice = newEarlyMintPrice;
+        emit EarlyMintPriceChanged(newEarlyMintPrice);
     }
 
-    function setBaseURI(string memory newBaseTokenURI) external onlyOwner {
+    function setBaseURI(string calldata newBaseTokenURI) external onlyOwner {
+        require(!permanentUrl, "NFTs minting finished, changing URL not possible");
```

Comment:
This could be handled using [Solidity Errors](https://docs.soliditylang.org/en/v0.8.10/contracts.html?highlight=errors#errors-and-the-revert-statement), instead of requires.

Reply:
Correct, but i said to keep a bit of a pattern? I think the focus is right now to get it out fast asap😅